parsl/launchers/launchers.py

```diff
@@ -34,30 +34,68 @@ class SingleNodeLauncher(Launcher):
     task_blocks to an integer or to a bash expression the number of invocations
     of the command to be launched can be controlled.
     """
-    def __call__(self, command, tasks_per_node, nodes_per_block):
+    def __call__(self, command, tasks_per_node, nodes_per_block, fail_on_any=False):
         """
         Args:
         - command (string): The command string to be launched
         - task_block (string) : bash evaluated string.
+        - fail_on_any: If True, return a nonzero exit code if any worker failed, otherwise zero;
+                       if False, return a nonzero exit code if all workers failed, otherwise zero.
```

Comment:
"Usually", things don't hang. We're here because, sometimes, they do.

Reply:
If at least 1 launched command ie manager is running, the job should not exit. The decision to terminate a job because the not all managers are running is upto strategy/executor, not the provider.

Reply:
Neither of those options cause the launcher to exit before all the launched commands complete. It is simply a question of what constitutes a  failure when multiple commands are involved.